.circleci/config.yml

```diff
@@ -0,0 +1,243 @@
+          command: |
+            cd example
+            flutter pub get
+      - run:
+          name: Build
+          command: |
+            cd example
+            flutter packages pub run build_runner build
+      - run:
+          name: Run tests
+          command: |
+            cd example
+            flutter drive --target=test_driver/app.dart --dart-define=API_KEY=$API_KEY
+jobs:
+  checkout_code:
+    description: "Checkout code"
+    docker:
+      - image: cirrusci/flutter:stable
+    steps:
+      - checkout
+      - persist_to_workspace:
+          paths: .
+          root: .
+
+  lint:
+    description: "Run static analysis for Flutter"
+    docker:
+      - image: cirrusci/flutter:stable
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Install Dependencies
+          command: flutter pub get
+      - run:
+          name: Build
+          command: flutter packages pub run build_runner build
+      - run:
+          name: Analyze code
+          command: flutter analyze lib
+
+  test:
+    description: "Run tests for Flutter"
+    docker:
+      - image: cirrusci/flutter:stable
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Install Dependencies
+          command: flutter pub get
+      - run:
+          name: Build
+          command: flutter packages pub run build_runner build
+      - run:
+          name: Run tests
+          command: flutter test
+
+  android-integration-test:
+    description: "Run integration tests for Flutter"
+    macos:
+      xcode: 11.3.1
+    steps:
+      - attach_workspace:
+          at: .
+      - setup-flutter
+      - android-sdk-dependencies
+      - create-launch-android-emulator
+      - install-and-build-flutter
+      - run-flutter-integration-tests
+
+  ios-integration-test:
+    description: "Run integration tests for Flutter"
+    macos:
+      xcode: 11.3.1
+    steps:
+      - attach_workspace:
+          at: .
+      - setup-flutter
+      - restore_cache:
+          name: Restore cocoapods specs and pods
+          key: v1-cocoapods-{{ checksum "./example/ios/Podfile.lock" }}-{{ arch }}
+      - simulator-dependencies
+      - run:
+          name: Open simulator
+          command: xcrun simctl boot "iPhone 11"
+      - install-and-build-flutter
+      - run-flutter-integration-tests
+      - save_cache:
+          name: Save cocoapods specs and pods cache
+          key: v1-cocoapods-{{ checksum "./example/ios/Podfile.lock" }}-{{ arch }}
+          paths:
+            - ./example/ios/Pods
+            - ~/.cocoapods
+
+workflows:
+  test:
+    jobs:
+      - checkout_code
+      - lint:
+          requires:
+            - checkout_code
+      - test:
+          requires:
+            - checkout_code
+  android-integration-test:
+    jobs:
+      - checkout_code
+      - android-integration-test:
+          requires:
+            - checkout_code
+  ios-integration-test:
```

Comment:
split in two workflows because I was running out of space in the partition. It looks like this helps

Reply:
we'll also get a separate check per workflow, right? And they can potentially run in parallel?

Reply:
Yes, I am not sure about the parallel, I think we might need to enable that, but I think potentially yes. I honestly like the two checks